### PR TITLE
[IMP] Improve detailed transfer performance

### PR DESCRIPTION
**Pull request by ghost**
Transfers are really slow in v8.0, due to the pickings remaining quantities being recomputed a lot of times.
This PR fixes the issue by recomputing the quantities only once

### CODE REVIEW

**Comment by beledouxdenis** [comment]
Looks correct to me ! I would nevertheless welcome @jco-odoo opinion :).

File: addons/stock/wizard/stock_transfer_details.py [hunk 1]
```diff
@@ -91,16 +92,17 @@ def do_detailed_transfer(self):
...
                 if prod.packop_id:
-                    prod.packop_id.write(pack_datas)
+                    prod.packop_id.with_context(self._context, no_recompute=True).write(pack_datas)
                     processed_ids.append(prod.packop_id.id)
+                    pickings |= prod.packop_id.picking_id
                 else:
                     pack_datas['picking_id'] = self.picking_id.id
                     packop_id = self.env['stock.pack.operation'].create(pack_datas)
                     processed_ids.append(packop_id.id)
+        pickings.do_recompute_remaining_quantities()
```
Comment (jco-odoo) [inline addons/stock/wizard/stock_transfer_details.py]: Not necessary as it is done in the do_transfer.

File: addons/stock/wizard/stock_transfer_details.py [hunk 2]
```diff
@@ -91,16 +92,17 @@ def do_detailed_transfer(self):
                     'owner_id': prod.owner_id.id,
                 }
                 if prod.packop_id:
-                    prod.packop_id.write(pack_datas)
+                    prod.packop_id.with_context(self._context, no_recompute=True).write(pack_datas)
```
Comment (jco-odoo) [inline addons/stock/wizard/stock_transfer_details.py]: This changes indeed everything.  I wonder if we could not even do without the recompute in all cases.

File: addons/stock/wizard/stock_transfer_details.py (hunk 1 above)
Comment (tgr-odoo) [inline addons/stock/wizard/stock_transfer_details.py]: That's even better then, I removed the unnecessary lines

**Comment by tgr-odoo** [comment]
@jco-odoo Can we have your final word about this?

**Comment by beledouxdenis** [comment]
Just a notice: I deployed the patch locally on a production server for beta-testing purposes :). Everything looks fine at the moment.

**Comment by jco-odoo** [comment]
I tested a simple case also and it works and the code looks ok.  Please
merge as this should be a great performance improvement.  (any idea how
much?)  Small remark: The self.with_context(self._context, ...) ->
self._context seems unnecessary

**Comment by tgr-odoo** [comment]
Yep, you're right, I removed it!

**Comment by tgr-odoo** [comment]
As for the performance improvement, I tried with a 100 moves picking.
Without the patch => 177 s of which 157 in the pack operation write
With the patch => 23 s of which 0.3 in the write

**Comment by beledouxdenis** [comment]
If I receive no negative feedback from the beta-testing by tomorrow, I will proceed to the merge in stable 8.0.

Regards

**Comment by beledouxdenis** [comment]
As you can see, I rebased the PR to make only one commit, and then merged this PR in stable Odoo 8.0.

Thomas, do I reply to the customer or I let you do it ?

**Comment by tgr-odoo** [comment]
Thanks Denis !

I'll contact the customer to let him know

**Comment by beledouxdenis** [comment]
Thank you !

**Pull request merged**